inst/include/child_model_simulation.hpp

```diff
@@ -13,54 +13,70 @@ void run_child_ageing(int time_step,
-          state_next.children.hc1_hiv_pop(hd, cat, a, s) +=
-            state_curr.children.hc1_hiv_pop(hd, cat, a - 1, s) * demog.survival_probability(a, s, time_step);
+    for (int a = 1; a < hc2_agestart; ++a) {
+      const auto survival_prob_per_group = survival_probability(a, s, time_step);
+      for (int hd = 0; hd < hc1DS; ++hd) {
+        for (int cat = 0; cat < hcTT; ++cat) {
+          hc1_hiv_pop_next(hd, cat, a, s) += hc1_hiv_pop_curr(hd, cat, a - 1, s) * survival_prob_per_group;
         }
-        for (int dur = 0; dur < ss.hTS; ++dur) {
-          state_next.children.hc1_art_pop(dur, hd, a, s) +=
-            state_curr.children.hc1_art_pop(dur, hd, a - 1, s) * demog.survival_probability(a, s, time_step);
+        for (int dur = 0; dur < hTS; ++dur) {
+          hc1_art_pop_next(dur, hd, a, s) += hc1_art_pop_curr(dur, hd, a - 1, s) * survival_prob_per_group;
         }
       }
     }
   }
 
-  for (int s = 0; s < ss.NS; ++s) {
-    for (int hd = 0; hd < hc_ss.hc1DS; ++hd) {
-      for (int hd_alt = 0; hd_alt < hc_ss.hc2DS; ++hd_alt) {
-        for (int cat = 0; cat < hc_ss.hcTT; ++cat) {
-          state_next.children.hc2_hiv_pop(hd_alt, cat, 0, s) += state_curr.children.hc1_hiv_pop(hd, cat, hc_ss.hc1_ageend, s) * demog.survival_probability(hc_ss.hc2_agestart, s, time_step) * cpars.hc_cd4_transition(hd_alt, hd);
+  for (int s = 0; s < NS; ++s) {
+    for (int hd = 0; hd < hc1DS; ++hd) {
+      for (int hd_alt = 0; hd_alt < hc2DS; ++hd_alt) {
+        const auto cd4_transition_prob = survival_probability(hc2_agestart, s, time_step) * hc_cd4_transition(hd_alt, hd);
+        for (int cat = 0; cat < hcTT; ++cat) {
+          hc2_hiv_pop_next(hd_alt, cat, 0, s) += hc1_hiv_pop_curr(hd, cat, hc1_ageend, s) * cd4_transition_prob;
         }
-        for (int dur = 0; dur < ss.hTS; ++dur) {
-          state_next.children.hc2_art_pop(dur, hd_alt, 0, s) += state_curr.children.hc1_art_pop(dur, hd, hc_ss.hc1_ageend, s) * demog.survival_probability(hc_ss.hc2_agestart, s, time_step) * cpars.hc_cd4_transition(hd_alt, hd);
+        for (int dur = 0; dur < hTS; ++dur) {
+          hc2_art_pop_next(dur, hd_alt, 0, s) += hc1_art_pop_curr(dur, hd, hc1_ageend, s) * cd4_transition_prob;
         }
       }
     }
   }
 
-  for (int s = 0; s < ss.NS; ++s) {
-    for (int a = (hc_ss.hc2_agestart + 1); a < pars.base.options.p_idx_fertility_first; ++a) {
-      for (int hd = 0; hd < hc_ss.hc2DS; ++hd) {
-        for (int cat = 0; cat < hc_ss.hcTT; ++cat) {
-          state_next.children.hc2_hiv_pop(hd, cat, a - hc_ss.hc2_agestart, s) +=
-            state_curr.children.hc2_hiv_pop(hd, cat, a - hc_ss.hc2_agestart - 1, s) *
-            demog.survival_probability(a, s, time_step);
+  for (int s = 0; s < NS; ++s) {
+    for (int a = (hc2_agestart + 1); a < p_idx_fertility_first; ++a) {
+      const auto survival_prob_per_group = survival_probability(a, s, time_step);
+      const auto hc2_age = a - hc2_agestart;
```

Comment:
I am take or leave on this one.

But if we are going to change, would like the name `hc2_age` to indicate that this is referring to an array index (rather than an age)

Reply:
im going to opt to not do this then because we are reducing the amount of proxy variables anyway and shortening `time_step`, i think itll be terse enough but if it does become an issue after the rewrite, i can always just do another PR that addresses all of these instances